pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -34,19 +33,21 @@ const (
 
 // Config for query_range middleware chain.
 type Config struct {
-	SplitQueriesByInterval time.Duration `yaml:"split_queries_by_interval" category:"advanced"`
-	AlignQueriesWithStep   bool          `yaml:"align_queries_with_step"`
-	ResultsCacheConfig     `yaml:"results_cache"`
-	CacheResults           bool `yaml:"cache_results"`
-	MaxRetries             int  `yaml:"max_retries" category:"advanced"`
-	ShardedQueries         bool `yaml:"parallelize_shardable_queries"`
-	CacheUnalignedRequests bool `yaml:"cache_unaligned_requests" category:"advanced"`
+	SplitQueriesByInterval        time.Duration `yaml:"split_queries_by_interval" category:"advanced"`
+	SplitInstantQueriesByInterval time.Duration `yaml:"split_instant_queries_by_interval" category:"experimental"`
+	AlignQueriesWithStep          bool          `yaml:"align_queries_with_step"`
+	ResultsCacheConfig            `yaml:"results_cache"`
+	CacheResults                  bool `yaml:"cache_results"`
+	MaxRetries                    int  `yaml:"max_retries" category:"advanced"`
+	ShardedQueries                bool `yaml:"parallelize_shardable_queries"`
+	CacheUnalignedRequests        bool `yaml:"cache_unaligned_requests" category:"advanced"`
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxRetries, "query-frontend.max-retries-per-request", 5, "Maximum number of retries for a single request; beyond this, the downstream error is returned.")
 	f.DurationVar(&cfg.SplitQueriesByInterval, "query-frontend.split-queries-by-interval", 24*time.Hour, "Split queries by an interval and execute in parallel. You should use a multiple of 24 hours to optimize querying blocks. 0 to disable it.")
```

Comment:
[nit] Now we should clarify this.

```suggestion
	f.DurationVar(&cfg.SplitQueriesByInterval, "query-frontend.split-queries-by-interval", 24*time.Hour, "Split range queries by an interval and execute in parallel. You should use a multiple of 24 hours to optimize querying blocks. 0 to disable it.")
```